standard/types.md

```diff
@@ -403,6 +403,187 @@ Implicit conversions are available from the `null` literal to `T?` ([§10.2.7](c
+
+As shown, a tuple can be initialized using a tuple literal (§tuple-literal-expressions-new-clause).
+
+An element need not have a name. An element without a name is unnamed.
+
+If a tuple declarator contains the type of all the tuple's elements, that set of types cannot be changed or augmented based on the context in which it is used; otherwise, element type information shall be inferred from the usage context. Likewise, for element names.
+
+A tuple's type can be declared explicitly. Consider the following declarations:
+
+```csharp
+(int, string) pair2 = (2, "Goodbye"); 
+(int code, string message) pair3 = (2, "Goodbye");
+(int code, string) pair4 = (2, "Goodbye"); 
+(int, string message) pair5 = (2, "Goodbye"); 
+(int code, string) pair6 = (2, message: "Goodbye");    // Warning: can't give a name to the second element
+(int code, string) pair7 = (newName: 2, "Goodbye");    // Warning: can't change the name of element code
+```
+
+The type of `pair2` is `(int, string)` with unknown element names. Similarly, the type of `pair3` is `(int, string)` but this time with the element names `code` and `message`, respectively. For `pair4` and `pair5`, one element is named, the other not.
+
+In the case of `pair6`, the second element is declared as being unnamed, and any attempt to provide a name in an initializing context shall be ignored. Likewise, for any attempt to change an element's name, as in the case of `pair7`.
+
+A tuple's type can be wholly inferred from the context in which it is used. Consider the following declarations:
+
+```csharp
+var pair10 = (1, "Hello"); 
+var pair11 = (code: 1, message: "Hello");
+var pair12 = (code: 1, "Hello");
+var pair13 = (1, message: "Hello");
+```
+
+The type of `pair10` is inferred from the initializer's tuple literal, as `(int, string)` with unknown element names. Similarly, the type of `pair11` is inferred from the initializer’s tuple literal, as `(int, string)` but this time with the element names `code` and `message`, respectively. For `pair12` and `pair13`, the element types are inferred, and one element is named, the other not.
+
+Element names within a tuple type shall be distinct.
+
+> *Example*:
+>
+> ```csharp
+> (int e1, float e1) t = (10, 1.2);                 // Error: both elements have the same name
+> (int e1, (int e1, int e2) e2) t = (10, (20, 30)); // OK: element names in each tuple type are distinct
+> ```
+>
+> *end example*
+
+The name of any element in a partial type declaration shall be the same for an element in the same position in any other partial declaration for that type.
+
+> *Example*:
+>
+> ```csharp
+> partial class C : IEnumerable<(string name, int age)> { ... }
+> partial class C : IEnumerable<(string fullname, int)> { ... } // Error: names must be specified and the same
+> ```
+>
+> *end example*
+
+A tuple cannot be created with the `new` operator. However, the `new` operator can be used to create and initialize an array of tuple or a nullable tuple.
```

Comment:
`new Tuple<String, String>("a", "b")` is legal.  The rule is that a tuple type syntax cannot be used with the new operator.